roles/infra/roles/storages/tasks/main.yml

```diff
@@ -91,7 +91,7 @@
     username: "{{ storages[item.0.name][item.0.storage.type].username | default(omit) }}"
     password: "{{ storages[item.0.name][item.0.storage.type].password | default(omit) }}"
     port: "{{ storages[item.0.name][item.0.storage.type].port | default(omit) }}"
-    target: "{{ storages[item.0.name][item.0.storage.type].target | default(omit) }}"
+    target: "{% if storages[item.0.name][item.0.storage.type].target not in item.0.storage.volume_group.logical_units | map(attribute='target') | list %}{{ storages[item.0.name][item.0.storage.type].target | default(omit) }}{% else %}{{ omit }}{% endif %}"
```

Comment:
The commit message does not explain the issue:

    The issue was that the role was trying to edit the storage connection
    target when the storage domain has multiple connection targets.

What is the problem with editing the connection when there are multiple
connections?

    This patch checks if the target which user input is in the list
    of targets from the storage domain if so it will ignore it else it will edit it.

What do you mean by "ignoring the target"?

The task is called "Validate connections of storages" but I cannot understand
what this code is doing. I don't see any validation or editing.

What does ovirt_storage_connection does? add a connection?
What does it do when you don't specify a target? What does it do
when you specify a target?

The actual change is not unreadable:

        target: "{% if storages[item.0.name][item.0.storage.type].target not in item.0.storage.volume_group.logical_units | map(attribute='target') | list %}{{ storages[item.0.name][item.0.storage.type].target | default(omit) }}{% else %}{{ omit }}{% endif %}"

252 characters line?! and I don't understand the syntax. Can we extract this to
readable python function(s)?

Finally a comment is needed, explaning why special handling is needed for the target.

Reply:
> The commit message does not explain the issue:
> 
> ```
> The issue was that the role was trying to edit the storage connection
> target when the storage domain has multiple connection targets.
> ```
> 
> What is the problem with editing the connection when there are multiple
> connections?
> 
> ```
> This patch checks if the target which user input is in the list
> of targets from the storage domain if so it will ignore it else it will edit it.
> ```
> 
> What do you mean by "ignoring the target"?
> 
A few of your questions, Peter described in the bz https://bugzilla.redhat.com/show_bug.cgi?id=1943221 
I'll update the commit msg.

> The task is called "Validate connections of storages" but I cannot understand
> what this code is doing. I don't see any validation or editing.
Well, this is done in the module itself.
> 
> What does ovirt_storage_connection does? add a connection?
> What does it do when you don't specify a target? What does it do
> when you specify a target?

We use it because the storage domain cannot change much in the API, but the storage connection can change many parameters IP for NFS, paths etc. If we would remove it then it would remove the idempotency of the storage domain. 

> 
> The actual change is not unreadable:
> 
> ```
>     target: "{% if storages[item.0.name][item.0.storage.type].target not in item.0.storage.volume_group.logical_units | map(attribute='target') | list %}{{ storages[item.0.name][item.0.storage.type].target | default(omit) }}{% else %}{{ omit }}{% endif %}"
> ```
> 
> 252 characters line?! and I don't understand the syntax. Can we extract this to
> readable python function(s)?
> 
> Finally a comment is needed, explaning why special handling is needed for the target.

ACK the 252 characters line I'll split it into multiple lines, but no python function this would not be ansible best practice.